app/src/androidTest/java/org/mozilla/focus/activity/BrowsingHistoryTest.java

```diff
@@ -92,42 +106,170 @@ public void tearDown() {
+     * 5. History is deleted */
+    @Test
+    public void browsingWebsiteAndDeleteItsHistoryItem_deleteSuccessfully() {
+
+        final String targetUrl = browsingWebsiteAndOpenHistoryPanel();
+
+        // Click the first history item
+        onView(withId(R.id.browsing_history_recycler_view))
+                .check(matches(atPosition(1, hasDescendant(withText(targetUrl)))));
+
+        // Open target history item's action menu
+        onView(withId(R.id.browsing_history_recycler_view)).perform(
+                RecyclerViewActions.actionOnItemAtPosition(1, clickChildViewWithId(R.id.history_item_btn_more)));
+
+        // Click the delete button
+        onView(withText(R.string.browsing_history_menu_delete))
+                .inRoot(RootMatchers.isPlatformPopup())
+                .perform(click());
+
+        // Check if browsing history is cleared
+        onView(withText(R.string.browsing_history_empty_view_msg)).check(matches(isDisplayed()));
+
+    }
+
+
+    /**
+     * Test case no: his_004
+     * Test case name: Clear all history record
+     * Steps:
+     * 1. Follow steps in his_001
+     * 2. Tap "CLEAR BROWSING HISTORY"
+     * 3. Dialog shows up
+     * 4. Tap "CANCEL"
+     * 5. History is not changed
+     * 6. Tap "CLEAR BROWSING HISTORY"
+     * 7. Dialog shows up again
+     * 8. Tap "CLEAR"
+     * 9. History cleared */
+    @Test
+    public void clearBrowsingHistory_cancelAndClearWorkCorrectly() {
+
+        browsingWebsiteAndOpenHistoryPanel();
+
+        // Click the "Clear browsing history" button
+        onView(withId(R.id.browsing_history_btn_clear)).perform(click());
+
+        // Check if the dialog is showed
+        onView(withText(R.string.browsing_history_dialog_confirm_clear_message)).check(matches(isDisplayed()));
+
+        // Click cancel button
+        onView(withText(R.string.action_cancel)).inRoot(isDialog()).perform(click());
+
+        // Click the "Clear browsing history" button
+        onView(withId(R.id.browsing_history_btn_clear)).inRoot(isDialog()).perform(click());
+
+        // Check if the dialog is showed
+        onView(withText(R.string.browsing_history_dialog_confirm_clear_message)).check(matches(isDisplayed()));
+
+        // Click clear button
+        onView(withText(R.string.browsing_history_dialog_btn_clear)).inRoot(isDialog()).perform(click());
+
+        // Check if browsing history is cleared
+        onView(withText(R.string.browsing_history_empty_view_msg)).check(matches(isDisplayed()));
+
+    }
+
+    private String browsingWebsiteAndOpenHistoryPanel() {
+        final String targetUrl = webServer.url(PATH_SITE_1).toString();
+
+        // Click search field
+        onView(withId(R.id.home_fragment_fake_input)).perform(click());
+
+        // Enter target url
+        onView(withId(R.id.url_edit)).perform(replaceText(targetUrl), pressImeActionButton());
+
+        // Check if target url is loaded
+        IdlingRegistry.getInstance().register(loadingIdlingResource);
```

Comment:
Should we call unregister in the same method once onView is done?

Reply:
Already fixed in the latest commit.